upup/models/cloudup/_aws/network.yaml

```diff
@@ -33,16 +33,19 @@ route/0.0.0.0/0:
   internetGateway: internetGateway/{{ ClusterName }}
   vpc: vpc/{{ ClusterName }}
 
-{{ range $zone := .Zones }}
+{{ range $index, $zone := .Zones }}
 
 subnet/{{ $zone.Name }}.{{ ClusterName }}:
   vpc: vpc/{{ ClusterName }}
   availabilityZone: {{ $zone.Name }}
   cidr: {{ $zone.CIDR }}
   id: {{ $zone.ProviderID }}
+  shared: {{ SharedZone $index }}
 
+{{ if not (SharedZone $index) }}
```

Comment:
Heh, now I'm trying to figure out what prompted me to do this the convoluted way :)

The cleanup fix looks good to me.